ProcessLib/ComponentTransport/ComponentTransportFEM.h

```diff
@@ -183,20 +271,27 @@ class LocalAssemblerData : public ComponentTransportLocalAssemblerInterface
                 _process_data.retardation_factor(t, pos)[0];
 
             auto const& solute_dispersivity_transverse =
-                _process_data.solute_dispersivity_transverse(t, pos)[0];
+                medium.template value<double>(
+                    MaterialPropertyLib::transversal_dispersivity);
+
             auto const& solute_dispersivity_longitudinal =
-                _process_data.solute_dispersivity_longitudinal(t, pos)[0];
+                medium.template value<double>(
+                    MaterialPropertyLib::longitudinal_dispersivity);
 
             // Use the fluid density model to compute the density
+            // TODO (renchao): concentration of which component as the argument
+            // for calculation of fluid density
             vars[static_cast<int>(
                 MaterialLib::Fluid::PropertyVariableType::C)] = C_int_pt;
             vars[static_cast<int>(
                 MaterialLib::Fluid::PropertyVariableType::p)] = p_int_pt;
             auto const density = _process_data.fluid_properties->getValue(
```

Comment:
Here, the density calculation depends only on one of the concentrations. Is this correct?